lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -172,10 +177,17 @@ void UnifyAssumesPass::processAssertInst(CallInst &CI, AllocaInst &flag) {
   llvm::CallSite CS(&CI);
   IRBuilder<> B(bb);
   B.SetInsertPoint(&CI);
+  Value *conseq = CS.getArgument(0);
+  // remove instruction if verifier.assert(true)
+  if (auto *conseq_const = dyn_cast<llvm::Constant>(conseq)) {
+    if (!conseq_const->isZeroValue()) {
```

Comment:
apparently `isZeroValue` is for floating point numbers. 
We should use `isZero` of `ConstantInt`.

Sorry, should have spotted this earlier, but only looked at LLVM code now before merging.

Reply:
Ok. I made this change.